src/poly.rs

```diff
@@ -0,0 +1,187 @@
+//! Polynomial over GF(256)
+//!
+//! adapted from https://github.com/mersinvald/reed-solomon-rs
+
+#[derive(Copy)]
+pub struct Polynom {
+    array: [u8; ::POLYNOMIAL_MAX_LENGTH],
+    length: usize,
+    dirty: bool,
+}
+
+impl Polynom {
+    #[inline]
+    pub fn new() -> Polynom {
+        Polynom {
+            array: [0; ::POLYNOMIAL_MAX_LENGTH],
+            length: 0,
+            dirty: false,
+        }
+    }
+
+    #[inline]
+    pub fn with_length(len: usize) -> Polynom {
+        let mut p = Polynom::new();
+        p.length = len;
+        p
+    }
+
+    #[inline]
+    pub fn set_length(&mut self, new_len: usize) {
+        let old_len = self.len();
+        self.length = new_len;
+        
+        if self.dirty && new_len > old_len {
+            for x in self.iter_mut().skip(old_len)
+                                    .take(new_len - old_len) 
+            {
+                *x = 0;
+            }
+        } else if new_len < old_len {
+            self.dirty = true;
```

Comment:
I don't see anywhere where the `dirty` flag is getting set to `false` again. Why not just ignore the bits further than `length`?